keps/sig-node/24-apparmor-ga/README.md

```diff
@@ -0,0 +1,682 @@
+  <!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required _prior to targeting to a milestone / release_.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in
+      [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and
+      SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for
+      publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to
+      mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once
+it is marked `implementable` should be approved by each of the KEP approvers. If
+any of those approvers is no longer appropriate than changes to that list should
+be approved by the remaining approvers and/or the owning SIG (or SIG-arch for
+cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every
+time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+This is a proposal to upgrade the AppArmor annotation on pods to a dedicated
+field, and graduate the feature to GA. This proposal aims to do the _bare
+minimum_ to clean up the feature, without blocking future enhancements.
+
+## Motivation
+
+AppArmor support has been added with Kubernetes v1.4 and is already in beta.
+Profiles have to be available on each node whereas the container runtime ensures
+that the profile is loaded when specified at pod or PSP level. Profiles can be
+specified per-container via the pod’s metadata annotation:
+
+```
+container.apparmor.security.beta.kubernetes.io/<container_name>: {unconfined,runtime/default,localhost/<profile>}
+```
+
+The feature has been more or less unchanged ever since. Also note that the
+addition predates feature gates or our modern concept of feature lifecycle. So,
+even though the annotations include `beta` in the key, this is entirely useable
+on any production GA cluster.
+
+The main motivation behind this KEP is to promote the AppArmor feature to GA.
+
+_NOTE: Seccomp was in a very similar state, but with some subtle differences.
+Promoting Seccomp to GA will be covered by a [separate
+KEP](/keps/sig-node/20190717-seccomp-ga.md)._
+
+### Goals
+
+- Declare AppArmor as GA
```

Comment:
Since it is currently beta and can be disabled via feature gate, a release in beta with the fields would make sense to me before promoting to GA.

Reply:
Okay, what do you think about this proposal @tallclair @pjbgf ?

Reply:
I am just not sure whether is there much benefit on changing the API whilst keeping it in beta, considering this feature kept the same API "stable" in beta for a few years.
I feel that changing the API without calling it GA may send the message that we have doubts about this feature/API and more changes may come down the line.

Going down that route, would that start the count down for deprecating the previous syntax or would we support both syntaxes until the feature GAs?